src/eventgrid/azext_eventgrid/_params.py

```diff
@@ -64,25 +69,58 @@ def load_arguments(self, _):
         c.argument('input_mapping_fields', arg_type=input_mapping_fields_type)
         c.argument('input_mapping_default_values', arg_type=input_mapping_default_values_type)
         c.argument('input_schema', arg_type=input_schema_type)
+        c.argument('odata_query', arg_type=odata_query_type)
 
     with self.argument_context('eventgrid topic') as c:
-        c.argument('topic_name', arg_type=name_type, help='Name of the topic', id_part='name', completer=get_resource_name_completion_list('Microsoft.EventGrid/topics'))
+        c.argument('topic_name', arg_type=name_type, help='Name of the topic.', id_part='name', completer=get_resource_name_completion_list('Microsoft.EventGrid/topics'))
+
+    with self.argument_context('eventgrid topic list') as c:
+        c.argument('topic_name', arg_type=name_type, help='Name of the topic.', id_part='name', completer=get_resource_name_completion_list('Microsoft.EventGrid/topics'))
+        c.argument('odata_query', arg_type=odata_query_type, id_part=None)
 
     with self.argument_context('eventgrid topic key list') as c:
-        c.argument('topic_name', arg_type=name_type, help='Name of the topic', id_part=None, completer=get_resource_name_completion_list('Microsoft.EventGrid/topics'))
+        c.argument('topic_name', arg_type=name_type, help='Name of the topic.', id_part=None, completer=get_resource_name_completion_list('Microsoft.EventGrid/topics'))
+
+    with self.argument_context('eventgrid domain create') as c:
+        c.argument('domain_name', arg_type=name_type, help='Name of the domain.', id_part='name', completer=get_resource_name_completion_list('Microsoft.EventGrid/domains'))
+
+    with self.argument_context('eventgrid domain delete') as c:
+        c.argument('domain_name', arg_type=name_type, help='Name of the domain.', id_part='name', completer=get_resource_name_completion_list('Microsoft.EventGrid/domains'))
+
+    with self.argument_context('eventgrid domain update') as c:
+        c.argument('domain_name', arg_type=name_type, help='Name of the domain.', id_part='name', completer=get_resource_name_completion_list('Microsoft.EventGrid/domains'))
```

Comment:
Instead of repeating this code three times, just use a scope of `eventgrid domain`, which will apply it to all commands and subgroups under the domain subgroup.

Reply:
Per our discussion, we needed to do this for domain to allow us to use different arguments and related names in domain topic scope. In specific, using --name for domain-topic-name (rather than domain-name which is used at domain scope) and domain-name for specifying the domain name (which is --name is domain scope).

Reply:
So a couple techniques to eliminate the duplication:

```Python
domain_name_type = CLIArgumentType(help='Name of the domain.', options_list='--domain-name', completer=get_resource_name_completion_list('Microsoft.EventGrid/domains'))

with self.argument_context('eventgrid domain') as c:
  c.argument('domain_name', domain_name_type, options_list=['--name', '-n'], id_part=name)

with self.argument_context('eventgrid domain topic') as c:
  c.argument('domain_name', domain_name_type, id_part=None)
```

And if you really needed to override per command (which is usually not the case):

```Python
for scope in ['create', 'update', 'delete']:
   with self.argument_context('eventgrid domain {}'.format(scope)) as c:
      c.argument('domain_name', …)
```

Reply:
Thx for the suggestion. Makes sense. Fixed.